src/main/java/org/jenkinsci/plugins/github_branch_source/HttpsRepositoryUriResolver.java

```diff
@@ -31,7 +31,10 @@
 
     @Override
     public String getRepositoryUri(String apiUri, String owner, String repository) {
-        return "https://" + hostnameFromApiUri(apiUri) + "/" + owner + "/" + repository + ".git";
+        if (apiUri.startsWith("https://")) {
```

Comment:
Indeed I see

    Fetching upstream changes from http://github.com/cloudbeers/PR-demo.git

in my log. So this is a regression.